packages/react-scripts/config/webpackDevServer.config.js

```diff
@@ -69,10 +68,16 @@ module.exports = function(proxy, allowedHost) {
     // in the Webpack development configuration. Note that only changes
     // to CSS are currently hot reloaded. JS changes will refresh the browser.
     hot: true,
+    // Use 'ws' instead of 'sockjs-node' on server since we're using native
+    // websockets in `webpackHotDevClient`.
+    transportMode: 'ws',
+    // Prevent a WS client from getting injected as we're already including
+    // `webpackHotDevClient`.
+    injectClient: false,
     // It is important to tell WebpackDevServer to use the same "publicPath" path as
     // we specified in the Webpack config. When homepage is '.', default to serving
     // from the root.
-    publicPath: shouldUseRelativeAssetPaths ? '/' : paths.servedPath,
+    publicPath: paths.publicUrlOrPath,
     // WebpackDevServer is noisy by default so we emit custom message instead
     // by listening to the compiler events with `compiler.hooks[...].tap` calls above.
     quiet: false,
```

Comment:
The only difference I see between your branch and upstream is this setting. When I forked in July, upstream had it as `false`, but after rebasing, upstream now has it as `true`. I don't feel strongly about it; if you find the additional logging useful it's find as is, but if you're just defaulting to what I had on my branch before, I'd be in favor of copying the decision they made to keep things in sync.